README.md

```diff
@@ -229,6 +243,253 @@ Benchee.run %{"Awesome stuff" => fn -> magic end }
+* Clearing the cache before a scenario
+
+#### before_each
+
+Is executed before each invocation of the benchmarking function/before every measurement. The input for `before_each` is either the `input` or what was returned by `before_scenario`. It works much like [`before_scenario`](#before_scenario), the only major difference being that it is invoked at another time. Most importantly, the function needs to return what the benchmarking function should receive as input.
+
+```elixir
+Benchee.run %{
+  "bar" => fn(input) -> bar(input) end # input will be 2 here
+}, inputs: %{
+  "input 1" => 1,
+  before_each: fn(input) -> input + 1 end
+}
+```
+
+When might this be useful?
+* Retrieving a record from the database and passing it on to the benchmarking function to do something(tm) without the retrieval from the database adding to the benchmark measurement
+* Busting caches so that all measurements are taken in an uncached state
+* Picking a random value from a collection and passing it to the benchmarking function for measuring performance with a wider spread of values
+
+### after_each
+
+Is executed directly after the invocation of the benchmarking function. It is passed the return value of the benchmarking function. Unlike the `before_*` functions the return value does not matter and isn't passed on. Both _global_ and _local_ `after_each` get the return value of the function.
+
+For `after_*` functions, the _local_ hook is invoked first, then the _global_ hook is invoked.
+
+```elixir
+Benchee.run %{
+  "bar" => fn(input) -> bar(input) end
+}, inputs: %{
+  "input 1" => 1,
+  after_each: fn(result) -> assert result == 42 end
+}
+```
+
+When might this be useful?
+* You can use this to run assertions that whatever your benchmarking function returns is truly what it should be (i.e. all "contestants" work as expected)
+* Busting caches/terminating processes setup in `before_each` or elsewhere
+
+### after_scenario
+
+Is executed after a scenario completed and is passed whatever the return value of the last `before_scenario` was, if there was none it is passed the specified `input`. Otherwise it works much like [`after_each`](#after_each).
+
+```elixir
+Benchee.run %{
+  "bar" => fn(input) -> bar(input) end
+}, inputs: %{
+  "input 1" => 1,
+  after_each: fn(_input) -> some_magic() end
+}
+```
+
+When might this be useful?
+* Bursting caches after a scenario completed
+* Deleting all the records in the database that the scenario just created
+* Terminating whatever was setup by `before_scenario`
+
+#### When does a hook happen? (Complete Example)
+
+Yes the whole hooks system is quite a lot to take in. Here is an overview when what hook is executed with what input depending on its previous hooks. In general, if some hook isn't specified it is safe to assume that instead just the value of the _previous_ hook is passed on.
+
```

Comment:
Don't say "safe to assume", just say what happens 😃 

"Hooks receive as input the return value of the previous hook."

Might it be useful to reference ExUnit's setup? The structure isn't the same, but the concept of passing data along from hook to another is similar.

Reply:
Not sure about ExUnit setups would actually have to read up :D